zio-http/src/main/scala/zhttp/http/middleware/Web.scala

```diff
@@ -171,6 +172,22 @@ private[zhttp] trait Web extends Cors with Csrf with Auth with HeaderModifier[Ht
       _ => middleware,
       _ => Middleware.identity,
     )
+
+  /**
+   * Removes the trailing slash from the path.
+   */
+  final def dropTrailingSlash: HttpMiddleware[Any, Nothing] =
+    Middleware.identity[Request, Response].contramap[Request](_.dropTrailingSlash)
+
+  /**
+   * Permanent redirect if the trailing slash is present in the request URL.
+   */
+  final def redirectTrailingSlash(permanent: Boolean): HttpMiddleware[Any, Nothing] =
+    Middleware.whenRequest(_.url.path.trailingSlash)(
+      dropTrailingSlash ++ Middleware.intercept[Request, Response](identity)((_, req) =>
+        Response.redirect(encode(req.dropTrailingSlash.url), isPermanent = permanent),
+      ),
+    )
```

Comment:
- Trailing slash should only be removed if there are no query params as per the RFC
- I think this middleware will still execute the HttpApp but ignore it's result.

Reply:
you are right, it is executing the  HttpApp.

Reply:
Done